clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/UpsertGoogleInternalHttpLoadBalancerAtomicOperation.java

```diff
@@ -0,0 +1,1074 @@
+        templateOpMap.put("image", instanceDescription.getImage());
+        templateOpMap.put("instanceType", instanceDescription.getInstanceType());
+        templateOpMap.put("credentials", credentials.getName());
+        templateOpMap.put("disks", instanceDescription.getDisks());
+        templateOpMap.put("instanceMetadata", instanceDescription.getInstanceMetadata());
+        templateOpMap.put("tags", instanceDescription.getTags());
+        templateOpMap.put("network", instanceDescription.getNetwork());
+        templateOpMap.put("subnet", instanceDescription.getSubnet());
+        templateOpMap.put("serviceAccountEmail", instanceDescription.getServiceAccountEmail());
+        templateOpMap.put("authScopes", instanceDescription.getAuthScopes());
+        templateOpMap.put("preemptible", instanceDescription.getPreemptible());
+        templateOpMap.put("automaticRestart", instanceDescription.getAutomaticRestart());
+        templateOpMap.put("onHostMaintenance", instanceDescription.getOnHostMaintenance());
+        templateOpMap.put("region", groupRegion);
+        templateOpMap.put("serverGroupName", groupName);
+
+        if (StringGroovyMethods.asBoolean(instanceDescription.getMinCpuPlatform())) {
+          templateOpMap.put("minCpuPlatform", instanceDescription.getMinCpuPlatform());
+        }
+
+        if (templateOpMap.containsKey("instanceMetadata")) {
+          Map<String, String> instanceMetadata = (Map) templateOpMap.get("instanceMetadata");
+          String regionLbStr = instanceMetadata.get(REGIONAL_LOAD_BALANCER_NAMES);
+          List<String> regionalLbs =
+              regionLbStr != null
+                  ? new ArrayList<>(Arrays.asList(regionLbStr.split(",")))
+                  : new ArrayList<>();
+          regionalLbs.add(loadBalancerName);
+          instanceMetadata.put(
+              REGIONAL_LOAD_BALANCER_NAMES,
+              regionalLbs.stream().distinct().collect(Collectors.joining(",")));
+
+          String backendsStr = instanceMetadata.get(REGION_BACKEND_SERVICE_NAMES);
+          List<String> bsNames =
+              backendsStr != null
+                  ? new ArrayList<>(Arrays.asList(backendsStr.split(",")))
+                  : new ArrayList<>();
+          bsNames.add(backendService.getName());
+          instanceMetadata.put(
+              REGION_BACKEND_SERVICE_NAMES,
+              bsNames.stream().distinct().collect(Collectors.joining(",")));
+        } else {
+          Map<String, String> instanceMetadata = new HashMap<>(2);
+          instanceMetadata.put(REGIONAL_LOAD_BALANCER_NAMES, loadBalancerName);
+          instanceMetadata.put(REGION_BACKEND_SERVICE_NAMES, backendService.getName());
+          templateOpMap.put("instanceMetadata", instanceMetadata);
+        }
+
+        AtomicOperationConverter converter =
+            atomicOperationsRegistry.getAtomicOperationConverter(
+                "modifyGoogleServerGroupInstanceTemplateDescription", "gce", ProviderVersion.v1);
```

Comment:
```suggestion
                "modifyGoogleServerGroupInstanceTemplateDescription", "gce");
```

Reply:
Thanks! I understand.

Reply:
Should I rebase this branch onto master?